### ISSUE TITLE
this.amOnPage is not a function

### ISSUE THREAD

**Issue by crustyratfink** [Codeception/CodeceptJS#1046]
#### What are you trying to achieve?
run basic tests with custom steps... alas, custom steps fail with:
`this.amOnPage is not a function` or some such similar thing that says it can't find a Nightmare function.

Defined some login function in steps_file.js, included in config
```
...
  "include": {
    "I": "./tests/steps_file.js"
  },
...
```
I think this has to do with moving the Nightmare helper to "node_modules" from "helpers" to get past an earlier error wherein I was having a ./helpers/Nightmare not found.

#### What do you get instead?
`this.amOnPage is not a function`

> Provide console output if related. Use `--verbose` mode for more details.

Feel like I'm beating a dead horse here, but...
```bash
  1) BUG-50: Assessment Hands-on
       I log in to Domain:
     this.amOnPage is not a function
```

> Provide test source code if related
Not super interesting:
```
Feature('BUG-50: Austria Hands-on');

Scenario('I log in to PLG', function (I) {
  I.login('username','password','Whatever');
});
```
and then in the steps_file.js
```
module.exports = function() {
  return actor({
    login: function(email, password, domain) {
      this.amOnPage(`/portal/domains/${domain}/#/login`);
...
```
### Details

* CodeceptJS version: 1.1.8
* NodeJS Version:
* Operating System:
* Protractor || WebDriverIO || Nightmare version (if related): `"codeceptjs-nightmare": "^1.2.0"`
* Configuration file:

```js
{
  "tests": "tests/*_test.js",
  "timeout": 10000,
  "output": "./output",
  "node_modules": {
    "Nightmare": {
      "url": "https://url.blah.net",
      "show": true,
      "restart": false
    }
  },
  "helpers": {
    "PermisoHelper": {
      "require": "./helpers/blah_helper.js"
    }
  },
  "include": {
    "I": "./tests/steps_file.js"
  },
  "bootstrap": false,
  "mocha": {},
  "name": "PermisoTests"
}
```

Long and short of this is that it seems that `this` isn't bound to the right thing.

Any thoughts?

**Comment by DavertMik** [comment]
Yeah, you have a very strange config. Where did you take that `node_modules` key? 0/o

Should be:

```json
  "helpers": {
    "Nightmare": {
      "url": "https://url.blah.net",
      "show": true,
      "restart": false
    },
    "PermisoHelper": {
      "require": "./helpers/blah_helper.js"
    }
  },
```

**Comment by crustyratfink** [comment]
OK, well, that's that, then. 

tl;dr I switched to Puppeteer, and it worked. 

In answer to your question about that node_modules key, I got it from:
https://github.com/Codeception/CodeceptJS/issues/443
Assumed it was some super secret config key that only the cool kids knew about.

Was never able to get Nightmare to work, global, local, helpers, node_modules, whatever.

**Comment by zhex900** [comment]
I have the same issue. Can anyone get Nightmare to work?

**Comment by crustyratfink** [comment]
Hey Jake, I was able to get it to work eventually by removing nightmare entirely and reinstalling it globally. Apparently, that node_modules thing was just one of those zany made up things, like chiropractic. I have no issues with Nightmare at this point. Be sure it's set as a helper in your config and installed globally.
